src/telemetry/stats.c

```diff
@@ -255,13 +255,22 @@ process_distributed_hypertable(BaseStats *stats, Form_pg_class class, const Hype
 static void
 process_continuous_agg(BaseStats *stats, Form_pg_class class, const ContinuousAgg *cagg)
 {
-	HyperStats *hyperstats = (HyperStats *) stats;
-	const Hypertable *ht = ts_hypertable_get_by_id(cagg->data.mat_hypertable_id);
+	CaggStats *cs = (CaggStats *) stats;
```

Comment:
I am not clear of what the advantage is to pass in a pointer to the `BaseStats` instead of passing in `CaggStats` directly. This would capture a lot of compile-time problems and not defer this to a runtime crash, if a mistake is made.